experimental/learn_descriptors/four_seasons_parser.hh

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+#include <stdio.h>
+
+#include <algorithm>
+#include <cstddef>
+#include <filesystem>
+#include <iostream>
+#include <opencv2/opencv.hpp>
+#include <string>
+#include <vector>
+
+#include "Eigen/Core"
+#include "Eigen/Geometry"
+#include "experimental/learn_descriptors/image_point.hh"
+
+namespace robot::experimental::learn_descriptors {
+class FourSeasonsParser {
+   public:
+    struct CameraCalibrationFisheye {
+        double fx, fy, cx, cy, k1, k2, k3, k4;
+    };
+    FourSeasonsParser(const std::filesystem::path& root_dir,
+                      const std::filesystem::path& calibration_dir);
+    cv::Mat load_image(const size_t idx) const;
+    const ImagePoint& get_image_point(const size_t idx) const { return img_pt_vector_[idx]; };
+    size_t num_images() const { return img_pt_vector_.size(); };
+    size_t size() const { return num_images(); };
+    const CameraCalibrationFisheye& get_camera_calibration() const { return cal_; };
+    const Eigen::Isometry3d& get_T_S_AS() const { return transforms_.T_S_AS; };
```

Comment:
consider a `liegroups::SE3` and calling this `S_from_AS()`. Also, what are `S` and `AS`?